E-Commerce-API/app.js

```diff
@@ -1,8 +1,175 @@
 const express = require("express");
 const app = express();
+var bodyParser = require("body-parser");
 // Your code to run the server should go here
 // Don't hardcode your DB password in the code or upload it to GitHub! Never ever do this ever.
 // Use environment variables instead:
 // https://www.codementor.io/@parthibakumarmurugesan/what-is-env-how-to-set-up-and-run-a-env-file-in-node-1pnyxw9yxj
+app.listener = app.listen(3000, function () {
+  console.log(`Server is listening on port ${3000}`);
+});
+const { Pool } = require("pg");
 
+const db = new Pool({
+  user: "mickeyhaile", // replace with you username
+  host: "localhost",
+  database: "cyf_ecommerce",
+  password: "",
+  port: 5432,
+});
+app.use(express.json());
+app.get("/products", (req, res) => {
+  db.query(
+    "SELECT p.product_name as name, pa.unit_price as price, s.supplier_name as supplierName FROM products p INNER JOIN product_availability pa on (p.id = pa.prod_id) INNER JOIN suppliers s on (pa.supp_id = s.id)",
+    (error, result) => {
+      if (error) {
+        console.error(error);
+        res.status(500).json({ error: "Internal Server Error" });
```

Comment:
I like the error handling, however careful with this as not every error is a 500 error; it could be a bad request or invalid credentials or something like that.